src/pint/models/binary_piecewise.py

```diff
@@ -0,0 +1,465 @@
+                    raise ValueError(
+                        f"Group index mismatch error. Check the indexes of T0X groups, make sure they match there are corresponding group ranges (PLB/PUB)"
+                    )
+            if len(ls_pub) > 0 and len(ls_A1X) > 0:
+                if len(np.setdiff1d(j_pub, j_A1X)) > 0:
+                    raise ValueError(
+                        f"Group index mismatch error. Check the indexes of A1X groups, make sure they match there are corresponding group ranges (PLB/PUB)"
+                    )
+            lb = [(getattr(self, tup[1])).value for tup in ls_plb]
+            ub = [(getattr(self, tup[1])).value for tup in ls_pub]
+
+            for i in range(len(lb)):
+                for j in range(len(lb)):
+                    if i != j:
+                        if max(lb[i], lb[j]) < min(ub[i], ub[j]):
+                            raise ValueError(
+                                f"Group boundary overlap detected. Make sure groups are not overlapping"
+                            )
+
+    def get_group_boundaries(self):
+        """Get a all pieces' date boundaries.
+        Returns
+        -------
+        list
+                np.array
+                        (length: toas) List of piecewise orbital parameter lower boundaries
+                np.array
+                        (length: toas) List of piecewise orbital parameter upper boundaries
+        """
+        # asks the object for the number of piecewise groups
+        return self.binary_instance.get_group_boundaries()
+
+    def which_group_is_toa_in(self, toa):
+        """Find the group a toa belongs to based on the boundaries of groups passed to BT_piecewise
+        Parameters
+        ----------
+        toa : toa
+           TOA/TOAs to check which group they're in
+        Returns
+        -------
+        np.array
+           str elements, look like ['0000','0001'] for two TOAs where one refences T0X_0000 or T0X_0001.
+        """
+        # asks the model what group a single TOA/list of TOAs
+        barycentric_toa = self._parent.get_barycentric_toas(toa)
+        return self.binary_instance.toa_belongs_in_group(barycentric_toa)
+
+    def get_number_of_groups(self):
+        """Get the number of piecewise parameters"""
+        return len(self.binary_instance.piecewise_parameter_information)
+
+    def get_group_indexes(self):
+        """Get all the piecewise parameter labels
+        Returns
+        -------
+        np.array
+                (length: number of piecewise groups) List of piecewise parameter labels e.g with pieces T0X_0000, T0X_0001, T0X_0003, returns [0,1,3]
+        """
+        group_indexes = []
+        for i in range(0, len(self.binary_instance.piecewise_parameter_information)):
+            group_indexes.append(
+                self.binary_instance.piecewise_parameter_information[i][0]
+            )
+        return group_indexes
+
+    def get_group_indexes_in_four_digit_format(self):
```

Comment:
Is this needed?  If you have the method above to return integers can't you just use:
```
[f"{int(...):04d" for x in ...]
```

Reply:
Yeah, this was an one of the functions I would use in the testing script. Does nothing essential, scrapped